### PR TITLE
chore: move to node 18 and dep updates

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Moved to Node 18
- Upgraded all non breaking deps
- Some have major changes that introduce breaking changes
- Pre looks on Vercel seem to have downed the bundle size though.

! NOTE !
Best to test it out on preview / local / extension
(I have a build)

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

**Review by idoshamun — APPROVED** [review]
Looks fine to me. Just double check the critical cases

**Comment by rebelchris** [comment]
For some reason the 2 test add `ht` to urls submitted.
I can't find the culprit, but since it's a low effort task I'll re-look at this later.

If anyone wants, feel free to have a look as well.

**Comment by capJavert** [comment]
> For some reason the 2 test add `ht` to urls submitted. I can't find the culprit, but since it's a low effort task I'll re-look at this later.
> 
> If anyone wants, feel free to have a look as well.

Just curious, which 2 test?

File: packages/shared/src/components/tooltips/ContextMenuItem.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ function ContextMenuItem({
         className={classNames('typo-callout', className)}
         onClick={onClick}
       >
-        <span className="flex items-center w-full">
+        <span className=" flex w-full items-center">
```
Comment (sshanzel) [inline packages/shared/src/components/tooltips/ContextMenuItem.tsx:36]: Just the extra space :P 
```suggestion
        <span className="flex w-full items-center">
```

**Review by sshanzel — APPROVED** [review]
Wow! This gets us a big step closer to upgrading our React as well. Also the bundle size improvement was amazing. Thank you!! 🙏

**Comment by rebelchris** [comment]
@capJavert If you click on the checks -> CircleCI you can see the 2 files that fail.
They are both the `submit link` and `submit article` modals, weird enough.

**Comment by capJavert** [comment]
@rebelchris yeah tnx for the explanation, I see 9 tests failing that is why I was asking. 

Just trying to understand, what do you mean by "`ht` added", what is `ht` specifically, like a string or?

**Comment by rebelchris** [comment]
@capJavert Yes it seems to append `ht` as a string to the input values.
I debugged it locally the input is just the URL, but after converting the form data we get `urlht` as response.

It did seem only to affect on test cases as testing it in browser seemed alright.

**Pull request closed without merging**